modules/grader/v3_util.go

```diff
@@ -0,0 +1,81 @@
+package grader
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"regexp"
+
+	"github.com/pegnet/pegnet/modules/factoidaddress"
+	"github.com/pegnet/pegnet/modules/opr"
+)
+
+// ValidateV2 validates the provided data using the specified parameters
+func ValidateV3(entryhash []byte, extids [][]byte, height int32, winners []string, content []byte) (*GradingOPR, error) {
+	if len(entryhash) != 32 {
+		return nil, NewValidateError("invalid entry hash length")
+	}
+
+	if len(extids) != 3 {
+		return nil, NewValidateError("invalid extid count")
+	}
+
+	if len(extids[1]) != 8 {
+		return nil, NewValidateError("self reported difficulty must be 8 bytes")
+	}
+
+	if len(extids[2]) != 1 || extids[2][0] != 3 {
+		return nil, NewValidateError("invalid version")
+	}
+
+	o, err := opr.ParseV2Content(content)
+	if err != nil {
+		return nil, NewDecodeError(err.Error())
+	}
+
+	if o.Height != height {
+		return nil, NewValidateError("invalid height")
+	}
+
+	// verify assets
+	if len(o.Assets) != len(opr.V2Assets) {
+		return nil, NewValidateError("invalid assets")
+	}
+	for _, val := range o.Assets {
+		if val == 0 {
+			return nil, NewValidateError("assets must be greater than 0")
+		}
+	}
+
+	if len(o.Winners) != 10 && len(o.Winners) != 25 {
```

Comment:
10 winners should be rejected as incorrect as V2 had already 25 winners

Reply:
You are correct. Good catch